compiler/rustc_middle/src/middle/limits.rs

```diff
@@ -48,10 +48,10 @@ fn update_limit(
                         .unwrap_or(attr.span);
 
                     let error_str = match e.kind() {
-                        IntErrorKind::Overflow => "`limit` is too large",
+                        IntErrorKind::PosOverflow => "`limit` is too large",
                         IntErrorKind::Empty => "`limit` must be a non-negative integer",
-                        IntErrorKind::InvalidDigit => "not a valid integer",
-                        IntErrorKind::Underflow => bug!("`limit` should never underflow"),
+                        IntErrorKind::InvalidDigit(_) => "not a valid integer",
+                        IntErrorKind::NegOverflow => bug!("`limit` should never underflow"),
```

Comment:
If we're removing the _underflow_ terminology we should probably do it in the message here too.